src/localizers/moment.js

```diff
@@ -8,12 +8,16 @@ function inSame12Hr(start, end){
   return (s <= 0 && e <= 0) || (s >= 0 && e >= 0)
 }
 
+function shouldTrimAMPM(start, local, culture) {
+  return /(AM|PM)$/i.test(local.format(start, 'LT', culture));
+}
+
 let dateRangeFormat = ({ start, end }, culture, local)=>
   local.format(start, 'L', culture) + ' — ' + local.format(end, 'L', culture)
 
 let timeRangeFormat = ({ start, end }, culture, local)=>
-  local.format(start, 'h:mma', culture) +
-    ' — ' + local.format(end, inSame12Hr(start, end) ? 'h:mm' : 'h:mma', culture)
+  local.format(start, 'LT', culture) +
+    ' — ' + local.format(end, (inSame12Hr(start, end) && shouldTrimAMPM(start, local, culture)) ? 'h:mm' : 'LT', culture)
```

Comment:
This basically leaves the AM/PM formatting alone. It would be great if just LT was accepted but then you still would have to trim AM/PM from the ranges which are within the same 24 hours.

The previous format would *force* everyone to submit to the AM/PM format which is really terrible localization. I changed very little compared to the original state. The only difference being that we that don't use the AM/PM system won't have it "forced".

> Right now this might display different things on either side of 12noon, and not just the AM/PM bits

Could you elaborate on this? I feel like I tested it thoroughly to do what I describe above. I.E. leaving the previous functionality as it was for users with AM/PM time and using the actual local format for non AM/PM.